pandas/types/cast.py

```diff
@@ -89,7 +89,8 @@ def trans(x):  # noqa
 
         if issubclass(dtype.type, np.floating):
             return result.astype(dtype)
-        elif dtype == np.bool_ or issubclass(dtype.type, np.integer):
+        elif (dtype == np.bool_ or issubclass(dtype.type, np.integer) and
```

Comment:
have to be careful about timedeltas because they are represented as integers, so new to be clear which dtype you have (and don't mix); this is more of an in general statement (not necessarily relevant to this particular question)